### DISCUSSION TITLE
Rename items that hApp devs encounter early on in their learning journey

### DISCUSSION THREAD

**Discussion by jost-s** [holochain/holochain#3385] [General]
There has repeatedly been feedback from hApp devs in forums, courses and through surveys that some of the essential names on the learning journey discovering Holochain are confusing and a hindrance. Some of these names are:

- DNA
- Coordinator (zome)
- Integrity (zome)
- Zome function/Zome call

Terminology preferably uses names that software developers are already familiar with. We want to come up with more suitable names for these things. We on the Holochain core team have brainstormed some replacement terms that we like to get the ball rolling, and now we want to share those more broadly to get feedback on our choices, and to uncover even better ideas others may have.

Here's our proposal for new names for these items:

| Existing term | Proposed replacement terms | Definition of the concept in terms of proposed replacement terms |
| -------- | -------- | ------- |
| DNA | a Schema | Modelset + Modifiers (does *not* include Coordinators) |
| DNA modifiers | Modifiers / Schema modifiers | Data which modifies the behavior of a Modelset to produce a Schema |
| DNA hash | Schema hash | The hash of a Schema which determines a unique network space |
| *(we currently lack a name for this concept)* | a Modelset / a set of Models | Ordered list of all Models in a Schema |
| Coordinator Zome | an API / a Coordinator / a Controller / a Microservice / a Business Logic Module / a Domain Logic Module / a Backend service | a Wasm containing functions that interact with a Cell's data | 
| Integrity Zome | a Model | a Wasm containing datatypes and validation rules | 
| Zome function | an API function / a Coordinator function / a Controller function / a Cell function | A function written in a Coordinator Wasm
| Zome call | an API call / a Coordinator call / a Controller call / a Cell call | Calling such a function |
| Zome | *We propose to remove this general concept entirely: we can always just refer to a Model or an API/Coordinator/Controller* | n/a |

### Notes

We suggest to remove "Zome" from the lexicon entirely. There is no need for a general term to talk about "a module which could be of type integrity or coordination". We can use the specific terms for each of those modules: "model" or "[coordinator (TBD)]". And if we need to talk about the details of its embodiment and execution as a Wasm module, we can call it just that: a Wasm module.

You'll notice that we haven't narrowed down the search for a replacement name for Coordinator zomes very much. We kinda like "coordinator" and "controller" but recognize there could be a much better term so **we're especially looking for feedback on that name**.

You'll also notice that we liked "Model" for the integrity layer and some variation on "Controller" for the coordination layer, which intentionally maps onto the [Model-View-Controller](https://en.wikipedia.org/wiki/Model%E2%80%93view%E2%80%93controller) design pattern, which we think could be a good conceptual fit for what those elements are, using a more familiar framework. Feedback on that premise is also welcome.

**Comment by mattyg** [comment]
Very supportive of this. It would dramatically improve the accessibility of holochain, and provider a clearer mental model for devs.

**Comment by guillemcordoba** [comment]
In general I like this, I think it fits. The one term I don't like is `Schema`. `Schema` feels to me like "a template for something that you conform to", but in this case it's more like "the instance of the game that we are playing", as it identifies the network. 

Honestly the closest things that ring true are "Rule Set" for what we now call "list of integrity zomes", and "Network" or "Game" for what we now call DNA. Not sure about this, can't find better terms. Saying things like "the network hash is the result of hashing the rule set, and potentially there are many agents in the network or none" feels right... 

Oh and shouldn't also the name "Cell" be changed together with all of these ones? Would feel weird otherwise.

**Reply by matthme** [reply]
I think DNA has actually been used ambiguously so far. For example I would not equate the `.dna` file to the "Network". In order for it to become the "Network", the dna file requires addition of network seed and specific properties (the dna file only defines the format/schema of the properties) and be installed into the conductor. So network for me is rather what we currently call "cell".

**Reply by jost-s** [reply]
@guillemcordoba Very interesting feedback. If `Schema` feels like that to you, I think that's we were hoping to convey. We were actually considering `Template` as well.

`Rule Set` is suitable too IMO.

**Reply by guillemcordoba** [reply]
@matthme Yeah totally! But already in my mind, a DNA already contains the DNA modifiers and identifies the network (the DNA hash). So I think we are on the same page as far as the concepts go, but I've been using the term DNA differently that the code and tooling implies, which points to the ambiguity that you are expressing.

What we call cell right now is not the network, the network is rather the collection of agents playing that game. 

@jost-s mmm. There is a difference between a "rule set" and the "instantiation of that rule set in context", which are the actual rules by which one game is played and identifies the network hash. "Schema" feels to me to be the first, but in the table you indicate that it already includes the modifiers, so that's why my confusion came up. 

Argh it's difficult to talk about changing names when the previous set of names is tricky!

**Reply by maackle** [reply]
A good name, like a joke, should work well without justification, but I'd justify Schema like so, at least to guide the search for even better names:

The integrity (model) layer is entirely about constraints. Its value is in defining what you *can't* do. It's nothing more than an intricate filter that data must pass through on its way to the DHT. So to me that is exactly "a template for something that you conform to" -- that conformity is what gives the network shape and meaning.

But it is not the network itself. Like @matthme points out, the rule set is not the game, the blueprint is not the building, and the schema is not the database. Other things need to exist for those things to come to life (players and an arena; builders and materials; a database program and a hard drive), but once the conditions are right, the dead thing (the DNA, the ruleset) gives rise to the living thing (the Network, the Game).

So I think it's good for the dead thing to not be given a "living" name. What it *is* is different from what it *makes possible*. I'm not attached to "schema" but I like names in the same direction.

**Reply by maackle** [reply]
It's also tricky because the distinction between "integrity zome" and "DNA" is subtle, it's just the "modifiers" (and the potential manyness of integrity zomes to one DNA). We have a name for an actual ruleset, and then a totally separate name for a ruleset that's been parameterized by a few values. Maybe we should bring these two names closer together. Maybe rethinking the term "modifiers" would help.

**Reply by maackle** [reply]
Also if we had finally implemented [zome-level modifiers](https://github.com/holochain/holochain/issues/909), then maybe there would have been no DNA-level modifiers, and what we now call a DNA would have just been a simple collection of integrity zomes, in which case we would hardly need a separate name for it.

For instance if we called an integrity zome a "schema" and we had schema modifiers, then the DNA might just be called a "schemata" (many schema glued together). We could have more exciting names than that but what do you think about using the same name for both, with the only distinction being the plurality?

**Reply by guillemcordoba** [reply]
Mmm the thing I don't like is the sentence "this network is identified by this schema hash", schemata hash doesn't feel good either... I don't think that we can't get away with only one name, because we are actually talking about concepts at two levels: development time concepts (DNA bundles, zomes, app bundle) and runtime ones (cells, dhts, apps) . The same app bundle installed twice with a different public key produces two different apps, same with what you are calling schema, specially with clones...

**Reply by pdaoust** [reply]
> `Schema` feels to me like "a template for something that you conform to"

do I understand correctly, then, that 'Schema' feels more appropriate for what's being called 'Modelset' above -- a template for the rules of a game that people can play? and then schema/modelset + modifiers = game? 'Network' doesn't feel quite right, because the modelset + modifiers bundle isn't yet a network until people start playing the game. Semantically, 'game' fits perfectly for me but relies too much on our tacit philosophy and culture to be accessible to newcomers. 'Network rule/model set' maybe?

But this works really well:

> the network hash is the result of hashing the rule set, and potentially there are many agents in the network or none

I'm going to try a sentence too: "If you want to use an existing rule set but need to create a new network, simply change its network seed -- it doesn't change the behaviour of the rule set, but still results in a unique network hash."

**Reply by guillemcordoba** [reply]
Yes totally! You are understanding perfectly Paul. Your point about network or game not being the right name is well taken, you are totally right. This is why I was talking earlier about development time concepts vs runtime concepts: for me (and it's also what the developer education materials follow), up until now, a DNA bundle was what's packaged in a .dna file, and a DNA is the actual instantiation of that with modifiers and all, identified by the DNA hash. Same with app: a .happ file is not an app, but an app bundle, and it's only when it's installed that an app is created. I think this difference is crucial and having good vocabulary around these is really important.

**Reply by pdaoust** [reply]
> Maybe we should bring these two names closer together. Maybe rethinking the term "modifiers" would help.

Let me try this language on, in the vein of my last message (sorry, there was a huge pile of conversation that happened between when I started my response and refreshed the page):

"The (rule set | model set | schema) and any optional modifiers are hashed together to produce the network hash."

What I find interesting here is that I don't really need a name for schema + modifiers in order for this to work. So perhaps our naming scheme doesn't need to be so total. But maybe that's not true, cuz we still need a file extension 😅 

What keeps coming up for me is that modifiers _are_ part of the rule set or schema, because they affect runtime behaviour. They're also like environment variables in a way -- outside the code but inherent to the way the code runs. The only thing that carries no semantic weight for the rule set is the network seed (and I guess network params like origin time). Maybe we just say things like (I'm in documentation mode here, trying to write as if I were writing for the Dev Portal to see if it works):

"When a schema is bundled or installed, you can specify optional runtime variables that can be accessed by the schema's code. Because they can affect runtime behaviour, they're considered part of the schema, and will affect the schema's network hash. You can also specify a network seed and certain network parameters, which also affect the network hash but don't affect runtime behaviour."

**Reply by pdaoust** [reply]
It also occurs to me that we could keep the `.dna` extension as a nice nod to history :)

**Reply by pdaoust** [reply]
BTW @guillemcordoba you make a very important distinction between the possibility of something existing and it actually existing. I still love 'cell' for that reason, but it's true that it feels out of place after we strip all the biological metaphors from everything else.

**Reply by matthme** [reply]
I think I agree with most that was said above and I like the notion of a "network hash" for what we currently call "dna hash" quite a bit since that's what it is after all - the hash that defines what network I'm participating in.
Reformulating what I think has been already said in essence: I think a lot of the confusion comes from the term "dna hash" since the way it has been used in practice was to denote the hash of the instantiated [cell minus agent key] rather than the hash of the dna in the "template" sense, i.e. zomes + modifier *structure*.

**Reply by maackle** [reply]
Hmm...game! Starting to like it. When I was first reading this, I was thinking of game as the actual live instance of playing as in "let's play a game of chess". But it can also mean "THE game of chess", as in what's written on the wikipedia page for Chess.

I feel more philosophical thoughts coming on, bear with me, I hope they may spur some ideas in others.

Running with what @guillemcordoba said about compile-time vs run-time, there at least 3 "times" of a game, similar to how software has compile-time, distribute-time, and run-time:

- There's the "compile-time" *idea of the game*: "do you know how to play chess?"
- There's the "distribute-time" *physical (spatial) representation of the game*: this is what you might go out to the store to buy, or make yourself, with a checkered board and pieces of particular shapes, colors, and materials, and probably a printout of the ruleset.
- And there's the "run-time" *temporal instantiation of the physical representation of the game*: when you actually set the pieces up and sit across from an opponent and play.

And these seem to map onto our concepts:

- The "compile-time" idea is the integrity zome set
- The "distribute-time" spatial representation is the DNA bundle, which is where Modifiers get introduced
- The "run-time" temporal instantiation is the Cell / Network

In that framing, the integrity zome(s) could definitely be called a "Ruleset" / "Schema" / "Model".

The DNA bundle is of course the packaged distributable thing, and the thing that makes it instantiable instead of just an idea, is the Modifiers, similar to how in order to actually play chess, you need to decide whether to make the pieces out of wood or plastic so you can touch them, and decide exactly what color of paint to use so you can see them. So I'd like a name for Modifiers that's less about modifying something existing, but actually the thing that takes something from an abstract idea into the possibility of a concrete reality, like the "implementation details". Something more like Environment, Setting, Context, etc.

As for the bundle itself, perhaps it can just be called a [whatever an integrity zome is called] + "bundle". SchemaBundle, ModelBundle, etc.? If not, then we are still searching for that term, and perhaps it is "Game"?

And the temporal instantiation is what you get when you get actual people playing the actual game. If we called a DNA a Game, then the Cell would almost certainly be called a Player. If we don't call it a game, then I don't know what to call it.

**Reply by pdaoust** [reply]
"environment and "context" work really well for me to describe the modifiers. Thinking about how they're currently used in software dev and adjacent fields:

* Environment variables are exactly how I think of DNA modifiers ✅ 
* Environment is also the OS, or some other container 🚩
* Context can mean the lexical scope that a lump of code operates in ✅ 
* Context can also mean a big object that carries information about the state of the system ✅ 

I like 'model bundle/set' and variations of that rather than a new name. Keeping it simple.

I really love the framing of a game. AND... I think it pulls us back into the world of a special cultural language that people have to grok in order to start using Holochain. For me the biggest concern about the term 'game' is that some people will immediately assume that Holochain is a game engine and stop reading if it doesn't fit their need.

**Comment by jost-s** [comment]
What are your thoughts on another name for "Coordinator"?

**Reply by zippy** [reply]
When we stick to the biological metaphor, the Coordinator Zome really should have been called the Cell Membrane.  It display's receptors to the outside (an API or the zome call definitions) and then, like any membrane which is two-layered, actually mediates and transforms what shows up on the out-side for the inside, i.e. trigger's instantiation of entries/like, and makes other zome calls or open up other receptors etc.  And it even makes sense that you can wrap different membranes around the same DNA to get different effects.

**Reply by mattyg** [reply]
Controller feels like a no-brainer to me. Just saying "a controller is a wasm bundle that depends on some models, exposes some external functions, and can be hot-swapped" describes the whole thing and is immediately understandable to a dev.

**Comment by ddd-mtl** [comment]
Just discovered this, and feel pretty frustrated about it. 
This means I will have to yet again waste time renaming everything and will waste brainpower adapting to the new terms, with no added value for the projects I'm involved with.
Its quite frustrating seeing attention being brought on to this instead of actually implementing features like zome-level modifiers that have been requested for years. Ugh.

I don't buy in the premise for doing name changes.
Lets face it, Holochain will always be confusing for newcomers, whatever the names.
Every specialized field needs its jargon. This allows the field's specialists to communicate more clearly and precisely about the things they do. Removing the jargon will most certainly diminish the specialists ability to communicate effectively.
Believing that jargon is the problem for newcomers to join a field is most likely a mistake. I'd say the issue is more about how and when to introduce jargon to newcomers.

So with all that said, here is my feedback:
- Please, don't change anything.
- Especially don't change DNA to Schema, as it doesn't make any sense to me.
- Zome is the most useful and clarifying term. Its heavily used in my codebases. Its not just a Wasm module as it implements holochain's DSL.
- Don't change "integrity zome" to Model. The funny thing, when you guys decided to go with 'coordinator zome' and 'integrity zome', at first,  I managed to not use those term much and went with 'zome' and 'model'. But with time, as my projects developed, it made more sense to use 'integrity zome', since its actually the data integrity layer which is different from the  data model. **The same data model can be used differently in two different integrity zomes**. Also, I remember Art describing holochain as a data integrity engine. So I believe it makes sense to keep that term as its good descriptive name of what it actually is, and is an important part of holochain.
- If you really need to change something, change 'coordinator zome' to 'controller zome', as coordinator still doesn't make any sense to me. And 'DNA Modifiers" could be changed to "DNA Parameters".

For more feedback, holo devs can still read what I wrote in holo's mattermost back in the days.
Thank you for reading.

**Comment by nphias** [comment]
Yeah I agree with @ddd-mtl and @zippy  ...  personally I feel Its not helpful to change all these terms now,  and will cause confusion among the community.  Some of you seem to disregard the fact that various people come to holochain with a basic understanding of nature/biology and these metaphors might be appreciated
    
I have always been drawn to holochain because of the biomimicry ... like object-orientated programming, i can visualize architecture because of the physical world references...   I ike refering to DNA, cells, zomes, roles, because i can have some tangible sensation about these things..   The last time the community changed some of the core terms .. we got rather ambiguous/generic names like Action and Record (Record conflict's with a typescript Record which doesnt help app devs).

My proposal would be to go in the other direction and introduce more biological terms including RNA, Receptors, populations and Proteins. :), while providing better documentation to explain associations and functions from the prespective of the "Cell".    

thank you

**Comment by pdaoust** [comment]
@ddd-mtl @nphias I hear you loudly and I want to hold this respectfully... I'm one of those people who was drawn to the biological metaphors, and I might not have given Holochain and Ceptr a second look if they weren't so chock-full of biology. I honestly love them and am willing to accept a fair bit of scientific inaccuracy when it comes to mapping the biological terms to the things that are happening in Holochain.

That said, as @jost-s pointed out, this inquiry is being driven by empirical findings. In 2022, we did a survey and were startled to find out that the biological language was a stumbling block for over a third of of respondents, and almost a quarter (averaged out over the words 'HDK', 'zome', and 'DNA') still either didn't understand what the words meant or demonstrated a non-functional understanding of them.

Now, the methodology of an open survey may be questionable, but this was after filtering out low-quality answers -- _and_ what's most important to me is that the survey was _biased toward the people who liked Holochain enough to (1) be following us and see the survey invitation and (2) take the time to respond to the survey_. Most were devs AFAIR. That's a high level of survivor bias in the responses.

It makes me wonder how many more people just gave up -- anecdotally, the number is at least 1, and this person I'm thinking of is no slouch -- CEO of Bluesky, back-end dev, DWeb veteran, with whom I had a difficult conversation about our 'frustratingly weird' language.

I think the point about the concepts of agent-oriented distributed systems being the biggest hurdle is very true. Given the feedback we've already gotten, the language is a surprisingly close runner-up. Hurdles that deliver value after you jump over them are worth keeping. I'm not sure whether the biological language is that -- for me and many of us here, it added value; for almost half of respondents (46%) it didn't.

I'm not going to hold strongly for dropping the biological language. I am going to hold strongly for developer experience -- and that also means the waves of new hApp developers that we want to join our community in the years ahead. Whether our survey results support that or not is a question worth considering.

Here's Carolyn's and my analysis and observations ~~; I haven't found the spreadsheet yet~~: https://docs.google.com/document/d/1kv_-BQGQlmjOeVo7-sF0s8b8xEBDcJ9P6VALviIPmlk/edit?usp=sharing There are also some comments from @maackle and @guillemcordoba that uncover things that have found new maturity in this recent discussion: that 'DNA', at the very least, is an ambiguous term that refers to two different things. Without precise terms, it's really hard to talk about Holochain in a way that devs can reason about.

[EDIT] Here are the anonymous survey results. https://docs.google.com/spreadsheets/d/1IuaxIKiGrKakvVISTit0aKeXVVBsJVnV7MoadxHwpXk/edit#gid=0 Raw data on sheet 1, sanitised data on sheet 2.

**Reply by ddd-mtl** [reply]
My main point is whatever new terms you use, you have no guarantee that its going to be less of a hurdle for new devs.
Whats guaranteed though is pain for current devs to rename and readapt to things.
Would be nice if Holo could A/B test terminology to see what works better.
Also, to make a fully informed decision, you'd want data/survey of actual devs making actual happs.

**Reply by nphias** [reply]
I totally agree that there are inaccuracies and the term "DNA" has been a source of confusion in the community.  How can you talk about DNA without talking about RNA?   Does it not make more sense to talk about DNA changes at the HDK level not the zome level?

I have always tired to explain holochain in biological terms. A cry from people in the survey was: "Where are the infographics?!"

![image](https://github.com/holochain/holochain/assets/17417820/b0186bde-d559-4492-bbc1-841c360df34f)

What If we redefined what we mean by DNA and RNA, proteins etc in the software and illustrated with something like this (above)?
I realize i am probably adding fuel to the jet engines of the anti-metaphorists right now.. but also feels nice to share this.. even if we lose the naming war..  the pro-metaphorists will retain proudly their ontological disposition

**Reply by ddd-mtl** [reply]
@nphias Some time ago, I almost convinced ppl at Holo to use Genome instead of DNA as its more accurate with the biology metaphor (dna would be the actual rust code in zomes), but I was told it doesnt work because devs are unfamiliar with this term. So, in a sense, whats preventing the use of biological metaphors is devs lack of knowledge in biology. I'd see it the other way: Using biological terms would have the positive side-effect of educating devs in biology.

**Reply by artbrock** [reply]
@nphias I kind of like your stack analogy, but it might work better to flip it upside down from your case. 

1. Rust (code in general) providing building block **proteins**
2. HDI/HDK providing assemble-able **RNA patterns**, 
3. Integrity zomes fleshing out those patterns into a coherent **DNA** of packaged code
4. Maybe the addition of the other hashed modifiers and network seeding info make up the **Genome**, 
5. Add coordinator zomes as the **regulatory proteins**, 
6. Add the UI/UX componentry for the full **environmental signaling system**

**Reply by artbrock** [reply]
And then it is more true to the biological stack and an opportunity, as @ddd-mtl points out, for devs to learn more biology.

**Reply by pdaoust** [reply]
@ddd-mtl all great points. Too easy to propose changes, assume that they're better, and then implement them without testing the effect. I'm at a bit of a loss for how to A/B test this -- we'd need to devise a test that filters out experienced devs, and show test subjects an alternative world of Holochain where the language isn't used. I wonder if there'd be a low-cost way to do this. I started out thinking we'd have to have an alternative HDK, but maybe just reworking the Core Concepts and doing a small usability test would be sufficient.

> Also, to make a fully informed decision, you'd want data/survey of actual devs making actual happs.

That's definitely an important consideration, although I'm concerned about recency bias there -- devs who have gotten to that point may already have forgotten most of the pain they experienced with the language. Whatever else is true about using the exercise to identify problems in our ontology all the way down to code, for me the renaming is all about flattening the learning curve for newcomers. It's my hope that the next wave(s) of hApp devs will outnumber those of us who already write hApps, and we have a potential opportunity to make that number larger than it otherwise could be.

Re: unfamiliarity with the biological terms, what I find interesting is that most of the critiques I've received are from people who are _very_ familiar with the terms, and try to project their understanding of cellular biology onto the labels we've used in Holochain, and find that the mapping really doesn't work consistently. I think the sweet spot is people like me who have a vague understanding of cellular biology, enough to kinda get the idea, but not enough to be able to quibble about details ;)

**Reply by pdaoust** [reply]
@nphias and I'm quite happy to proudly retain my ontological disposition in favour of the biological metaphors wherever they work -- and wherever I have enough space to clarify where they break down. (That's a point that Marcus, who led the in-person dev training, made to me: the metaphors don't have to be perfect, but you do have to explain where they're imperfect.)

That's why I feel like the proper place for these metaphors is in the documentation -- as an allegorical aside for people who really love this stuff and will grok it, rather than as the base language we impose everywhere. And I think that's a great place to invite people to understand biology better if they don't already get it -- not when they're already trying to understand hard concepts in distributed systems.

**Reply by maackle** [reply]
I agree that the biological names should be kept as metaphorical explanatory devices but not technical terms.

@nphias (I think, by way of the diagram posted in this thread) suggested that Coordinators are akin to part of RNA. @artbrock has suggested that Coordinators are akin to regulatory proteins. And elsewhere, @zippy has suggested that Coordinators are akin to cell membranes (@mjbrisebois calls them cell surface receptors, a part of the membrane). For all I know from my limited grasp on cellular biology, these are all perfectly valid metaphors. However, of course, they are all also perfectly divergent.

I acknowledge that these are not all actual proposals and the metaphors proposed in this thread are more like riffing on an idea, so I don't want to paint a picture that different people holding for biological terms are actually at odds with each other. However, to me this illustrates that the biological terms are *too rich*. They are so open to personal interpretation and poetic license. They create a beautiful and inspiring image in the mind of each individual, connecting the tech we're building to a much greater and much more advanced natural technology. But, more likely than not, the images created by these metaphors will be different for each person, because there are so many possible valid choices for mapping Holochain concepts onto biology. And thus I think these names fail in their primary job as technical terms, which is to convey a definite meaning that is shared by all without ambiguity. By the same token, they succeed spectacularly in creating an inspiring and motivating way to look at this tech that hints at all the possibilities and grand ambitions of it, which is why I also feel that they're better suited for motivational and persuasive documentation like the whitepaper.

As much as I see individuals disappointed at the prospect of current bio terms getting renamed, I could imagine that if we went all-in on biological naming, those same individuals could be disappointed that their own personal mapping from the vast world of biology to the comparatively tiny world of Holochain was not chosen. And isn't that disappointment pretty much guaranteed, given the 3-way divergence of just one little concept, "coordinator"? It's clear to me that there is no one true mapping from Holochain to biology, and even if I'm wrong, arriving at that true mapping will have to be a whole other debate in itself.

Aside from being an engineer I am also an artist, and as such I appreciate the way that through poetic metaphor each individual arrives at their own inspiration and their own truth, and also appreciate that Holochain represents a system which fundamentally allows for divergence of worldview. But as an engineer I do not appreciate that poetic license extending to the tools with which I build and communicate with my fellow builders.

**Reply by maackle** [reply]
I also acknowledge that I could be totally wrong about my argument about "regulatory protein" being something fundamentally different from "cell surface receptor", and maybe someone firmly grounded in cellular biology would immediately see the connection and it would click for them. I could also be wrong that there is no one true mapping from biology onto Holochain, maybe it's obvious to biologists. I'm curious what the biologists among us think.
